File: python/docmost/services/comment_resolution_api.py
"""Generated API client."""
from __future__ import annotations

from typing import Any, Optional

import httpx

from docmost.http_client import DocmostHttpClient
from docmost.types import ApiResponse
from docmost.models import ResolveCommentDto


class CommentResolutionApi:
    """Comment Resolution API."""

    def __init__(self, http: DocmostHttpClient) -> None:
        self._http = http

    def resolve_comment(self, request: ResolveCommentDto) -> ApiResponse[Any]:
        return self._http.post("comments/resolve", body=request.model_dump(by_alias=True, exclude_none=True))

File: go/docmost/services/comment_resolution_api.go
package services

import (
	"context"

	"github.com/ggpo1/docmost-sdk/go/internal/httpclient"
	"github.com/ggpo1/docmost-sdk/go/docmost/models"
)

// CommentResolutionAPI implements Comment Resolution API.
type CommentResolutionAPI struct {
	http *httpclient.HTTPClient
}

func NewCommentResolutionAPI(http *httpclient.HTTPClient) *CommentResolutionAPI {
	return &CommentResolutionAPI{http: http}
}

func (a *CommentResolutionAPI) ResolveComment(ctx context.Context, request *models.ResolveCommentDto) (*httpclient.ApiResponse, error) {
	return a.http.Post(ctx, "comments/resolve", request)
}

